app/services/authentication/providers/provider.rb

```diff
@@ -0,0 +1,60 @@
+module Authentication
+  module Providers
+    # Authentication provider
+    class Provider
```

Comment:
base provider class, consumes the authentication payload and has some method that the actual providers need to implement